src/drop.js

```diff
@@ -22,5 +22,5 @@ var _xdrop = require('./internal/_xdrop');
  *      R.drop(3, [1,2,3,4,5,6,7]); //=> [4,5,6,7]
  */
 module.exports = _curry2(_dispatchable('drop', _xdrop, function drop(n, list) {
-  return n < list.length ? _slice(list, n) : [];
+  return n <= 0 ? list : _slice(list, n);
```

Comment:
@megawac: If our API is meant to deal mostly with immutable data, why should Ramda take the hit of cloning?  If one user needs to worry about reference equality, then she will need to do defensive cloning as necessary, but other users shouldn't have to pay the perf costs of that.

I guess I'm not particularly worried about 

```javascript
var a = [4,5,6];
var n = random(-5, 5);

R.take(n, a).push(1,2,3)
```

when 

```javascript
R.concat(R.take(n, a), [1, 2, 3])
```

actually _is_ consistent.